### ISSUE TITLE
use fasta format as input format for unipept command line tools

### ISSUE THREAD

**Issue by pdawyndt** [unipept/unipept#261]
command line tools that operate from a list of proteins or peptides expect a single entry per line from stdin; in addition to this, it would be a good idea to also accept FASTA format as input format

I would not add an extra parameter to specify the input format, but plain and fasta format could be recognized by the fact that the first (non-whitespace) character of a fasta file is the ">" character

_[Original issue](https://github.ugent.be/unipept/unipept/issues/261) by @pdawyndt on Wed Apr 30 2014 at 14:17._
_Closed by @bmesuere on Wed Apr 22 2015 at 14:26._

**Comment by pdawyndt** [comment]
if FASTA format is used as the input format, FASTA format could also be used as the default output format instead of plain text; this would allow to use the FASTA identifier to link input records to output records; this is needed to link output records to input records if there is not a 1:1 between both; for a lot of the tools there is a 1:0..n mapping between input records and output records

possible implementation, using `seq.fasta` as an input file

```
>SRR029686.100108_1_255_-
GVLLYYLVGVSMFILICWAMITDVIDDTEVKTGSRSDGTVYSIYSFCRKLGQALSSGLTGVLLSMVGYSTATAFDTPVVN
GIYN
>SRR029686.100018_1_255_+
GDMHPNENLSVKDLHNEKYQGIRPAVGYPSLTPTNLSNFVLNELLDMKQIGYFFRTENGMMXPHRPSCPRPGTVLPVPAP
TSLA
>SRR029686.100017_1_255_-
LCALPIPCCVSQFITYFNTYPRGTFTELARFHAGKALFLDTPEPRLDQSSTYQAIQQLQMFMEYFPNSTKKQEAQDMIFA
LQDK
```

implementation if fasta input/output

```
$ prot2pept < seq.fasta
>SRR029686.100108_1_255_-:0
GVLLYYLVGVSMFILICWAMITDVIDDTEVK
>SRR029686.100108_1_255_-:1
TGSR
>SRR029686.100108_1_255_-:2
SDGTVYSIYSFCR
>SRR029686.100108_1_255_-:3
K
>SRR029686.100108_1_255_-:4
LGQALSSGLTGVLLSMVGYSTATAFDTPVVNGIYN
>SRR029686.100018_1_255_+:0
GDMHPNENLSVK
>SRR029686.100018_1_255_+:1
DLHNEK
>SRR029686.100018_1_255_+:2
YQGIRPAVGYPSLTPTNLSNFVLNELLDMK
>SRR029686.100018_1_255_+:3
QIGYFFR
>SRR029686.100018_1_255_+:4
TENGMMXPHRPSCPRPGTVLPVPAPTSLA
>SRR029686.100017_1_255_-:0
LCALPIPCCVSQFITYFNTYPR
>SRR029686.100017_1_255_-:1
GTFTELAR
>SRR029686.100017_1_255_-:2
FHAGK
>SRR029686.100017_1_255_-:3
ALFLDTPEPR
>SRR029686.100017_1_255_-:4
LDQSSTYQAIQQLQMFMEYFPNSTK
>SRR029686.100017_1_255_-:5
K
>SRR029686.100017_1_255_-:6
QEAQDMIFLQDK
```
- implementation if csv output: add `input record,output record` as extra fields to the output
- implementation if xml output: add `input="xxx" output="yyy"` as arguments to record tag

output record is always numbered (starting from zero) per input record; input record is either identified by fasta-identifier or by record number (starting from zero); plain list is special case of csv output


_[Original comment](https://github.ugent.be/unipept/unipept/issues/261#issuecomment-9091) by @pdawyndt on Thu May 01 2014 at 09:37._

**Comment by pdawyndt** [comment]
the above numbering scheme for input and output records, would also render the inclusion of for example the peptide from the input as one of the fields in the output unnecessary for some use cases, and at least it the peptide should not be sent back from the server (which improves bandwidth somewhat)


_[Original comment](https://github.ugent.be/unipept/unipept/issues/261#issuecomment-9127) by @pdawyndt on Fri May 02 2014 at 20:43._

**Comment by bmesuere** [comment]
Prot2pept and peptfilter should re-output fasta header,
unipept should add extra column for csv output.


_[Original comment](https://github.ugent.be/unipept/unipept/issues/261#issuecomment-9661) by ghost on Wed May 14 2014 at 14:37._

**Comment by pdawyndt** [comment]
`@tvwillem` although user wants fasta input results to be converted into fasta output results, the command line tools could still bundle input into chunks that are larger than a single fasta record, to reduce the number of server requests


_[Original comment](https://github.ugent.be/unipept/unipept/issues/261#issuecomment-9662) by @pdawyndt on Wed May 14 2014 at 14:47._

**Comment by bmesuere** [comment]
Basic support for FASTA input has been added in unipept 0.5.0. 


_[Original comment](https://github.ugent.be/unipept/unipept/issues/261#issuecomment-9847) by ghost on Mon May 19 2014 at 15:26._

**Comment by pdawyndt** [comment]
`@tvwillem` I have reopened this issue, just as a reminder that I still need to test it properly; this will be a major improvement to the user interface of the command line (most candidate users work with FASTA files)


_[Original comment](https://github.ugent.be/unipept/unipept/issues/261#issuecomment-9855) by @pdawyndt on Mon May 19 2014 at 18:19._

**Comment by bmesuere** [comment]
This was fixed in version 0.6.1


_[Original comment](https://github.ugent.be/unipept/unipept/issues/261#issuecomment-16456) by @bmesuere on Wed Apr 22 2015 at 14:26._
